### PR TITLE
tests: adds 6LoWPAN receiving test

### PR DESCRIPTION
**Pull request by Yonezawa-T2**
This test sends fragmented 6LoWPAN packets with header compression to itself and checks the received packet. I made this for #5033.

### CODE REVIEW

**Comment by Yonezawa-T2** [comment]
Here is a handcrafted hex dump for Wireshark:
https://gist.github.com/Yonezawa-T2/1f780296babe467e3e08cfa241de36fb
The "Encapsulation Type" is "IEEE 802.15.4 Wireless PAN with FCS not present".

**Comment by miri64** [comment]
Just thinking if you can use `gnrc_pktdump` to output the reassembled packet in various stages (so you can check it with pexpect). The same you could also do for your `gnrc_ipv6_ext` test.

**Comment by Yonezawa-T2** [comment]
`gnrc_pktdump` is great. Updated.
Depends on #5504, will rebase when it is merged.

**Comment by miri64** [comment]
#5504 was merged, please rebase.

**Comment by Yonezawa-T2** [comment]
Rebased & squashed.

File: tests/gnrc_sixlowpan/main.c [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+    gnrc_netif_get(ifs);
+
+    struct {
+        gnrc_netif_hdr_t netif_hdr;
+        char src[8];
+        char dst[8];
+    } netif_hdr = {
+        .src = { 0x02, 0x00, 0x00, 0xFF, 0xFE, 0x00, 0x00, 0x02 },
+        .dst = { 0x02, 0x00, 0x00, 0xFF, 0xFE, 0x00, 0x00, 0x01 },
```
Comment (miri64) [inline tests/gnrc_sixlowpan/main.c]:
```
Building application "gnrc_sixlowpan" for "native" with MCU "native".

/home/martine/Repositories/RIOT-OS/RIOT/tests/gnrc_sixlowpan/main.c: In function ‘_send_packet’:
/home/martine/Repositories/RIOT-OS/RIOT/tests/gnrc_sixlowpan/main.c:61:36: error: overflow in implicit constant conversion [-Werror=overflow]
         .src = { 0x02, 0x00, 0x00, 0xFF, 0xFE, 0x00, 0x00, 0x02 },
                                    ^
/home/martine/Repositories/RIOT-OS/RIOT/tests/gnrc_sixlowpan/main.c:61:42: error: overflow in implicit constant conversion [-Werror=overflow]
         .src = { 0x02, 0x00, 0x00, 0xFF, 0xFE, 0x00, 0x00, 0x02 },
                                          ^
/home/martine/Repositories/RIOT-OS/RIOT/tests/gnrc_sixlowpan/main.c:62:36: error: overflow in implicit constant conversion [-Werror=overflow]
         .dst = { 0x02, 0x00, 0x00, 0xFF, 0xFE, 0x00, 0x00, 0x01 },
                                    ^
/home/martine/Repositories/RIOT-OS/RIOT/tests/gnrc_sixlowpan/main.c:62:42: error: overflow in implicit constant conversion [-Werror=overflow]
         .dst = { 0x02, 0x00, 0x00, 0xFF, 0xFE, 0x00, 0x00, 0x01 },
                                          ^
cc1: all warnings being treated as errors
/home/martine/Repositories/RIOT-OS/RIOT/Makefile.base:60: recipe for target '/home/martine/Repositories/RIOT-OS/RIOT/tests/gnrc_sixlowpan/bin/native/gnrc_sixlowpan/main.o' failed
make[1]: *** [/home/martine/Repositories/RIOT-OS/RIOT/tests/gnrc_sixlowpan/bin/native/gnrc_sixlowpan/main.o] Error 1
/home/martine/Repositories/RIOT-OS/RIOT/tests/gnrc_sixlowpan/../../Makefile.include:259: recipe for target 'all' failed
```

I think its recommendable to have these fields of type `uint8_t` (see #5497)

Comment (Yonezawa-T2) [inline tests/gnrc_sixlowpan/main.c]: Amended.

**Comment by miri64** [comment]
I'll give my ACK as soon as I have tested it locally :)

File: tests/gnrc_sixlowpan/main.c [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    gnrc_netreg_entry_t dump_ipv6 = { NULL, GNRC_NETREG_DEMUX_CTX_ALL, gnrc_pktdump_pid };
+    gnrc_netreg_entry_t dump_udp = { NULL, GNRC_NETREG_DEMUX_CTX_ALL, gnrc_pktdump_pid };
+    gnrc_netreg_entry_t dump_udp_61616 = { NULL, 61616, gnrc_pktdump_pid };
+
+    gnrc_netreg_register(GNRC_NETTYPE_SIXLOWPAN, &dump_6lowpan);
+    gnrc_netreg_register(GNRC_NETTYPE_IPV6, &dump_ipv6);
+    gnrc_netreg_register(GNRC_NETTYPE_UDP, &dump_udp);
+    gnrc_netreg_register(GNRC_NETTYPE_UDP, &dump_udp_61616);
+
+
```
Comment (miri64) [inline tests/gnrc_sixlowpan/main.c]: There be whitespace ;-)

**Comment by miri64** [comment]
Apart from the introduced whitespace-error: ACK and go, as soon as Murdock likes it.

**Comment by Yonezawa-T2** [comment]
Fixed whitespace error.

**Pull request merged**